engine/crates/engine-v2/src/sources/graphql/request/prepare.rs

```diff
@@ -29,7 +33,8 @@ impl PreparedGraphqlOperation {
         operation_type: OperationType,
         plan: PlanWalker<'_>,
     ) -> Result<PreparedGraphqlOperation, Error> {
-        let mut ctx = QueryBuilderContext::default();
+        let mut ctx = QueryBuilderContext::new(plan.subgraph_id());
```

Comment:
IMO it would be better to provide `subgraph_id` to the `build()` function instead of adding subgraph_id & endpoint to the PlanWalker. The caller is `GraphqlResolver` which has `definition: GraphqlRootFieldResolverDefinition<'_>` with everything you need already. Same comment for the `PreparedFederationEntityOperation` underneath. That would also guarantee you to you have `SubgraphId` instead of an option.